app/services/trs/register_for_trn.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Trs
+  class RegisterForTrn
+    include ServicePattern
+
+    attr_reader :trainee
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      return unless FeatureService.enabled?(:integrate_with_trs)
+
+      submit_trn_request
+      trainee.submit_for_trn! if trainee.can_submit_for_trn?
+      trn_request
+    end
+
+  private
+
+    def submit_trn_request
+      trn_request.response = response
+      trn_request.requested!
+      Trainees::SetSlugSentAt.call(trainee:)
+    rescue Client::HttpError => e
+      trn_request.response = { error: e }
+      trn_request.failed!
+      Sentry.capture_exception(e)
+    end
+
+    def trn_request
+      @trn_request ||=
+        ::Dqt::TrnRequest.create_with(request_id:)
```

Comment:
Sense check, we going to rename the namespace/module/table?

Reply:
The remaining DQT for now so we don't have to update all the existing DQT code